packages/angular/cli/commands/update-impl.ts

```diff
@@ -28,11 +33,96 @@ const npa = require('npm-package-arg');
 
 const oldConfigFileNames = ['.angular-cli.json', 'angular-cli.json'];
 
-export class UpdateCommand extends SchematicCommand<UpdateCommandSchema> {
+export class UpdateCommand extends Command<UpdateCommandSchema> {
   public readonly allowMissingWorkspace = true;
 
-  async parseArguments(_schematicOptions: string[], _schema: Option[]): Promise<Arguments> {
-    return {};
+  private workflow: NodeWorkflow;
+
+  async initialize() {
+    this.workflow = new NodeWorkflow(
+      new virtualFs.ScopedHost(new NodeJsSyncHost(), normalize(this.workspace.root)),
+      {
+        packageManager: await getPackageManager(this.workspace.root),
+        root: normalize(this.workspace.root),
+      },
+    );
+
+    this.workflow.engineHost.registerOptionsTransform(
+      validateOptionsWithSchema(this.workflow.registry),
+    );
+  }
+
+  async executeSchematic(
+    collection: string,
+    schematic: string,
+    options = {},
+  ): Promise<{ success: boolean; files: Set<string> }> {
+    let error = false;
+    const logs: string[] = [];
+    const files = new Set<string>();
+
+    const reporterSubscription = this.workflow.reporter.subscribe(event => {
+      // Strip leading slash to prevent confusion.
+      const eventPath = event.path.startsWith('/') ? event.path.substr(1) : event.path;
+
+      switch (event.kind) {
+        case 'error':
+          error = true;
+          const desc = event.description == 'alreadyExist' ? 'already exists' : 'does not exist.';
+          this.logger.warn(`ERROR! ${eventPath} ${desc}.`);
```

Comment:
Shouldn't an error use the `.error`?

Reply:
It should also be nice that instead of duplicating the method a function is created to share this logic.

Reply:
It probably should be an error.

I would like to do a larger refactor of the schematic command at some point which would pull the schematic execution elements into a standalone class that could then be leveraged as needed.